packages/core/src/utils.js

```diff
@@ -767,6 +767,32 @@ export function retrieveSchema(schema, rootSchema = {}, formData = {}) {
     return resolveCondition(schema, rootSchema, formData);
   }
 
+  if (resolvedSchema.properties) {
+    const properties = {};
+
+    Object.entries(resolvedSchema.properties).forEach(entries => {
+      const propName = entries[0];
+      const propSchema = entries[1];
+      const rawPropData = formData && formData[propName];
+      const propData = isObject(rawPropData) ? rawPropData : {};
+      const resolvedPropSchema = retrieveSchema(
+        propSchema,
+        rootSchema,
+        propData
+      );
+
+      properties[propName] = resolvedPropSchema;
+
+      if (
+        propSchema !== resolvedPropSchema &&
+        resolvedSchema.properties !== properties &&
+        properties !== null
```

Comment:
I think `properties !== null` check is redundand, isn't?

Reply:
Thanks, you're right - this was a holdover from when I was addressing an issue that caused the patch to fail [this test](https://github.com/rjsf-team/react-jsonschema-form/blob/d87a88b99c1e9f793b414cb32d412ea9b59293ec/packages/core/test/utils_test.js#L1096)